doc/reference/bluetooth/mesh/shell.rst

```diff
@@ -0,0 +1,730 @@
+
+Prerequisites
+*************
+
+The Mesh Shell subsystem depends on the :ref:`bluetooth_mesh_models_cfg_cli` and :ref:`bluetooth_mesh_models_health_cli` models.
+
+Mesh Shell application
+**********************
+
+The Mesh Shell subsystem is most easily used through the Mesh Shell application under ``tests/bluetooth/mesh_shell``.
+See :ref:`shell_api` for information on how to connect and interact with the shell application.
+
+Basic usage
+***********
+
+The Mesh Shell subsystem adds a single ``mesh`` command, which holds a set of sub-commands. Every time the device boots up, make sure to call ``mesh init`` before any of the other Mesh Shell commands can be called::
+
+	uart:~$ mesh init
+
+Provisioning
+============
+
+The mesh node must be provisioned to become part of the network. This is only necessary the first time the device boots up, as the device will remember its provisioning data between reboots.
+
+The simplest way to provision the device is through self-provisioning. To provision the device with the default network key and address ``0x0001``, execute::
+
+	uart:~$ mesh provision 0 0x0001
+
+Since all mesh shell nodes use the same values for the default network key, this can be done on multiple devices, as long as they're assigned non-overlapping unicast addresses. Alternatively, to provision the device into an existing network, the unprovisioned beacon can be enabled with ``mesh pb-adv on`` or ``mesh pb-gatt on``. The beacons can be picked up by an external provisioner, which can provision the node into its network.
+
+Once the mesh node is part of a network, its transmission parameters can be controlled by the general configuration commands:
+
+* To set the destination address, call ``mesh dst <addr>``.
+* To set the network key index, call ``mesh netidx <NetIdx>``.
+* To set the application key index, call ``mesh appidx <AppIdx>``.
+
+By default, the transmission parameters are set to send messages to the provisioned address and network key.
+
+Configuration
+=============
+
+By setting the destination address to the local unicast address (``0x0001`` in the ``mesh provision`` command above), we can perform self-configuration through any of the :ref:`bluetooth_mesh_shell_cfg_cli` commands.
+
+A good first step is to read out the node's own composition data::
+
+	uart:~$ mesh get-comp
+
+This prints a list of the composition data of the node, including a list of its model IDs.
+
+Next, since the device has no application keys by default, it's a good idea to add one::
+
+	uart:~$ mesh app-key-add 0 0
+
+Message sending
+===============
+
+With an application key added (see above), the mesh node's transition parameters are all valid, and the Mesh Shell can send raw mesh messages through the network.
+
+For example, to send a Generic OnOff Set message, call::
+
+	uart:~$ mesh net-send 82020100
```

Comment:
I guess it would be important to specify how a value is broken up into bytes in the correct byte order, right?

Reply:
Added note on endianness.